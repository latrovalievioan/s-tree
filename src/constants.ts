// ^ - Match the beginning of the line
// ( - Start capture group 1
// ((?!(/|\\s))) - Negative lookahead assertion for / or \\s(empty space)
// . - Match any char
// ) - End capture group 1
// * - Match 0 or more of capture group 1
// $ - Match the end of the line
export const OBJECT_NAME_REGEX = '^((?!(/|\\s)).)*$';
export const DELETE_CONFIRMATION_STRING = 'DELETE';
