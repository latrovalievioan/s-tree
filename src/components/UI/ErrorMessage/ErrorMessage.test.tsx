import { render, screen, cleanup } from '@testing-library/react';
import { ErrorMessage } from '.';
import { DEFAULT_ERROR_MESSAGE } from '@/constants';

describe(ErrorMessage, () => {
  afterEach(() => {
    cleanup();
  });

  it('Should render the component', () => {
    render(<ErrorMessage />);
    const errorMessage = screen.getByTestId('error-message');

    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toBeVisible();
  });

  it('Should show the correct default string', () => {
    render(<ErrorMessage />);
    const errorMessage = screen.getByTestId('error-message');

    expect(errorMessage).toHaveTextContent(DEFAULT_ERROR_MESSAGE);
  });

  it('Should show the passed message', () => {
    const testMessage = 'this is a test message';
    render(<ErrorMessage message={testMessage} />);
    const errorMessage = screen.getByTestId('error-message');

    expect(errorMessage).toHaveTextContent(testMessage);
  });

  it('Matches snapshot', () => {
    render(<ErrorMessage />);
    const errorMessage = screen.getByTestId('error-message');

    expect(errorMessage).toMatchSnapshot();
  });
});
