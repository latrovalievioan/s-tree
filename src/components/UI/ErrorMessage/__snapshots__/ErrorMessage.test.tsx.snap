// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage > Matches snapshot 1`] = `
<div
  class="error"
  data-testid="error-message"
>
  There was an unexpected error, please refresh the page or try again.
</div>
`;
